internal/watcher/nginx_process_parser.go

```diff
@@ -56,28 +56,70 @@ func NewNginxProcessParser() *NginxProcessParser {
 	}
 }
 
-func (npp *NginxProcessParser) Parse(ctx context.Context, processes []*model.Process) []*v1.Instance {
-	instanceList := []*v1.Instance{}
+// cognitive complexity of 16 because of the if statements in the for loop
+// don't think can be avoided due to the need for continue
+// nolint: revive
+func (npp *NginxProcessParser) Parse(ctx context.Context, processes []*model.Process) map[string]*v1.Instance {
+	instanceMap := make(map[string]*v1.Instance)   // key is instanceID
+	workers := make(map[int32][]*v1.InstanceChild) // key is ppid of process
 
 	nginxProcesses := filterNginxProcesses(processes)
 
 	for _, nginxProcess := range nginxProcesses {
-		// Here we are determining if the nginxProcess is a master process.
-		// NGINX worker processes are ignored.
-		_, ok := nginxProcesses[nginxProcess.Ppid]
-		if !ok {
+		// Here we are determining if the nginxProcess is a worker process
+		if masterProcess, ok := nginxProcesses[nginxProcess.Ppid]; ok {
```

Comment:
Dunno if this is a big hassle, but could we perhaps rename the fields in `Process` so that they are capitalized? Tripped up several times here confusing `Pid` and `Ppid`, so think this might help with that slightly.
```go
type Process struct {
	Pid  int32
	Ppid int32
	Name string
	Cmd  string
	Exe  string
}

```
becomes

```go
type Process struct {
	PID  int32
	PPID int32
	Name string
	Cmd  string
	Exe  string
}
```
Initialisms and acronyms should be capitalized per Go conventions: https://google.github.io/styleguide/go/decisions.html#initialisms

I know this isn't part of your PR, so can do it in a follow-up PR.